### DISCUSSION TITLE
Multiple body parameters's requireds are not consistent with pydantic BaseModel?

### DISCUSSION THREAD

**Discussion by bastienexertier** [fastapi/fastapi#9665] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

class Item(BaseModel):
	item_id:str

class Model(BaseModel):
	item_1:Item
	item_2:Item|None
	item_3:Item|None=None

app = FastAPI()

@app.post('/1')
def post_normal_body(model:Model):
	return {}

@app.post('/2')
def post_embed_body(item_1:Item, item_2:Item|None, item_3:Item|None=None):
	return {}
```

### Description

Hello,
In this example I created two endpoints taking as parameters 3 Item objects:
- `item_1` required
- `item_2` optional without a default value
- `item_3` optional with a default value

However one uses an explicit pydantic BaseModel and the other one uses FastAPI [Multiple body parameters](https://fastapi.tiangolo.com/tutorial/body-multiple-params/?h=embed#multiple-body-parameters)

This means the two endpoints accept this:
```json
{
  "item_1": {
    "item_id": "string"
  },
  "item_2": {
    "item_id": "string"
  },
  "item_3": {
    "item_id": "string"
  }
}
```
However I find an inconsistency with optional fields:
- ✔️ Both of them REQUIRE `item_1`.
- ✔️ Both of them DO NOT REQUIRE `item_3`
- ❌ Only the Multiple body parameters version REQUIRES `item_2`

This is easily testable with /docs

Is this expected behavior? I would have thought both of them would NOT require `item_2`.

Thank you!

### Operating System

Windows

### FastAPI Version

0.96.0

### Python Version

3.10.2

**Comment by bastienexertier** [comment]
Looking the FastAPI source code, this looks to be on the edge between FastAPI and Pydantic

**Comment by wu-clan** [comment]
I believe it depends on the working mode of pydantic, while multiparameter is based on python.

**Comment by wu-clan** [comment]
The working mode of pydantic will change in 2.0 and will be the same as the multi-parameter mode, meaning that `item_2: Item | None` will be a required item even if it is None

**Reply by wu-clan** [reply]
pydantic-v1: https://docs.pydantic.dev/1.10/usage/models/#required-optional-fields
pydantic-v2: https://docs.pydantic.dev/dev-v2/usage/models/#required-optional-fields

**Reply by bastienexertier** [reply]
But looking at this line, it seems that the required value is computed on FastAPI side and given to pydantic ModelField so I'm wondering if changes on pydantic side will after this
https://github.com/tiangolo/fastapi/blob/32935103b12b1548117abef0b4af9dd883898308/fastapi/dependencies/utils.py#L467

**Reply by wu-clan** [reply]
Looking here, it seems that empty fields with no value will never be allowed to be passed in
https://github.com/tiangolo/fastapi/blob/32935103b12b1548117abef0b4af9dd883898308/fastapi/dependencies/utils.py#L388

**Reply by wu-clan** [reply]
If I find in the pydantic-v1 source code that `item_2: Item | None` returns something other than `Undefined` by default, I think my point will be correct

**Reply by wu-clan** [reply, marked as answer]
The v1 optional field seems to always return None instead of Undefined, so when using pydantic, None is passed in by default, which is equivalent to passing in None manually when using multiple parameters
https://github.com/pydantic/pydantic/blob/90326480ff1061f690aa193bd112cd17a6d90a70/pydantic/validators.py#L573

**Reply by bastienexertier** [reply]
Thank you it's clear now! Hopefully pydantic v2 will bring the consistency between the two!
